dissect/disc/disc.py

```diff
@@ -0,0 +1,756 @@
+                        logger.warning(
+                            "%s format is not available for this disc. Falling back to %s.", preference.value, fmt.value
+                        )
+                    self.iso_format = fmt
+                    break
+
+        if self.iso_format is None:
+            raise ValueError("Could not select format for disc.")
+
+        # Now that we know how we're going to treat the disc, we re-load the root record and the logical block size.
+        root_record = c_iso.iso_directory_record(self.primary_volume.root_directory_record)
+        self.logical_block_size = self.primary_volume.logical_block_size
+        if self.iso_format == ISOFormats.JOLIET:
+            self.name_encoding = "utf-16-be"
+
+        # Wrap the root_record in the appropriate class for the selected file format.
+        self.root_record = self.make_record(root_record)
+
+    @property
+    def primary_volume(self) -> Instance:
+        """Return the primary volume corresponding with the selected format to treat this disc as."""
+        return self.primary_volumes[self.iso_format]
+
+    @property
+    def path_table(self) -> dict[str, int]:
+        """Parse the path table and create a dict of paths and the corresponding extent of their directory record."""
+        if self._path_table is not None:
+            return self._path_table
+
+        self._path_table = dict()
+
+        self.fh.seek(self.primary_volume.type_l_path_table * self.logical_block_size)
+        path_table_bytes = self.fh.read(self.primary_volume.path_table_size)
+
+        offset = 0
+        index = 1
+        entries: dict[int, str] = dict()
+
+        while offset < self.primary_volume.path_table_size:
+            entry = c_iso.iso_path_table_entry(path_table_bytes[offset:])
+            if index == 1:
+                entries[1] = "/"
+                self._path_table["/"] = entry.extent_location
+            else:
+                entry_name = entry.name.decode(self.name_encoding)
+                parent_path = entries[entry.parent_dir_no]
+                seperator = "/" if parent_path != "/" else ""
+                entry_path = parent_path + seperator + entry_name
+
+                entries[index] = entry_path
+                self._path_table[entry_path] = entry.extent_location
+
+            entry_size = len(entry)
+            offset += entry_size
+            if entry_size % 2 != 0:
+                offset += 1
+
+            index += 1
+
+        return self._path_table
+
+    def make_record(self, record: Instance) -> DirectoryRecord:
+        """Wrap a iso_directory_record structure in the appropriate python representation based on the selected format
+        of this disc."""
+        if self.iso_format == ISOFormats.ROCKRIDGE:
+            return RockRidgeDirectoryRecord(self, record)
+        return DirectoryRecord(self, record)
+
+    def get_entry(self, path: str, use_path_table: bool = False) -> DirectoryRecord:
```

Comment:
```suggestion
    def get(self, path: str, use_path_table: bool = False) -> DirectoryRecord:
```
More consistent with other Dissect libraries.